src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -466,6 +463,10 @@ is ready so it starts to read that file (go back two windows and
 
     emitMetrics(currWindowStartTime);
 
+    if (appContext != null && !clusterDetailsEventsProcessor.getNodesDetails().isEmpty()) {
```

Comment:
should we add a few comments here to explain why we want to do this check ?

Reply:
^ I was confused by this too, I had to look into AppContext to figure out why a comment would be nice

Reply:
Good point. Will do